plugins/importer/src/main/java/com/managemc/importer/reader/EssentialsReader.java

```diff
@@ -0,0 +1,60 @@
+package com.managemc.importer.reader;
+
+import com.managemc.importer.reader.model.EssentialsMute;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.file.*;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class EssentialsReader implements PunishmentReader<EssentialsMute> {
+
+  private final File essentialsPluginFolder;
+
+  public EssentialsReader(File essentialsPluginFolder) {
+    this.essentialsPluginFolder = essentialsPluginFolder;
+  }
+
+  @Override
+  public List<EssentialsMute> read() throws Exception {
```

Comment:
This is where all of our reading logic lives. This class reads EssentialsX's mute store. Note how we use `Files.walkFileTree`, because we don't want to load potentially millions of files into memory at once. Essentials has one file per player that logs in and/or gets muted.